src/host.cpp

```diff
@@ -838,6 +923,34 @@ namespace rhost {
             unblock_message_loop();
         }
 
+        void handle_binary_message(ws_connection_type::message_ptr msg) {
+            if (binary_message_queue.size() != 1) {
+                fatal_error("Unexpected incoming client message - host was expecting a single pending binary message.");
+            }
+
+            message incoming = binary_message_queue.front();
```

Comment:
Looks like this check should also be under queue mutex, since it reads something that can be mutated from another thread.